sdk/core/azure-core/tests/async_tests/test_pipeline_async.py

```diff
@@ -219,4 +219,42 @@ def send(*args):
     policies = [AsyncRetryPolicy(), NaughtyPolicy()]
     pipeline = AsyncPipeline(policies=policies, transport=None)
     with pytest.raises(AzureError):
-        await pipeline.run(HttpRequest('GET', url='https://foo.bar'))
\ No newline at end of file
+        await pipeline.run(HttpRequest('GET', url='https://foo.bar'))
+
+@pytest.mark.asyncio
+async def test_add_custom_policy():
+    class BooPolicy(AsyncHTTPPolicy):
+        def send(*args):
+            raise AzureError('boo')
+
+    class FooPolicy(AsyncHTTPPolicy):
+        def send(*args):
+            raise AzureError('boo')
+
+    boo_policy = BooPolicy()
+    foo_policy = FooPolicy()
+    client = AsyncPipelineClient(base_url="test", per_call_policies=boo_policy)
+    policies = client._pipeline._impl_policies
+    assert boo_policy in policies
+
+    client = AsyncPipelineClient(base_url="test", per_call_policies=[boo_policy])
+    policies = client._pipeline._impl_policies
+    assert boo_policy in policies
```

Comment:
We never assert the location/ordering of the custom policies in the pipeline?

Reply:
Yes we can. But hard-coding the location will make our tests fragile (adding policy into our default pipeline may break the test). I will re-think about it. :)

Reply:
We just need to test that the index is < or > than the index of the Retry policy right? The exact index shouldn't matter.

Reply:
Yes. This is exactly I am trying to do. :)

Reply:
Added.